components/MapRoute.js

```diff
@@ -235,41 +251,89 @@ class MapRoute extends React.Component {
     );
   }
 
+  renderNavigationButton() {
+    const { showNavigationButton } = this.props;
+    const { destinationReached, isNavigation } = this.state;
+
+    if (destinationReached || isNavigation || !showNavigationButton) {
+      return null;
+    }
+
+    return (
+      <View style={styles.confirmContainer}>
+        <TouchableOpacity
+          style={styles.confirmButton}
+          onPress={this.startNavigation}
+        >
+          <View style={styles.drawerItem}>
+            <Ionicons
+              name="ios-checkmark-circle-outline"
+              size={30}
+              color="#ffffff"
+              style={styles.drawerItemIcon}
+            />
+            <Text style={styles.buttonText}>Start Navigation</Text>
+          </View>
+        </TouchableOpacity>
+      </View>
+    );
+  }
+
+  renderRoute() {
+    const { isMapReady, coordinates } = this.state;
+    if (!isMapReady || coordinates.length === 0) {
+      return null;
+    }
+    return (
+      <MapView.Polyline
+        coordinates={coordinates}
+        strokeWidth={5}
+        strokeColor="blue"
+      />
+    );
+  }
+
+  renderMarker() {
+    const { isMapReady, coordinates } = this.state;
+    if (!isMapReady || coordinates.length === 0) {
+      return null;
+    }
+    return (
+      <MapView.Marker
+        coordinate={coordinates[coordinates.length - 1]}
+      />
+    );
+  }
+
   onMapReady = () => {
     this.setState({ isMapReady: true });
   };
 
   render() {
-    const { coordinates, focusedLocation, isMapReady } = this.state;
-
     return (
       <View style={styles.container}>
         <MapView
+          provider="google"
           style={styles.map}
-          initialRegion={focusedLocation}
           showsUserLocation
           followsUserLocation={Platform.OS === 'ios'}
```

Comment:
`followsUserLocation={Platform.OS === 'ios'}` can be removed, as now `checkUserLocation` navigates to the user position for iOS as well.

Reply:
Done.